docs/types.rst

```diff
@@ -840,6 +849,81 @@ Of course, you can also directly access the members of the struct without
++------------------------------------------------+---------------------------------------------------------+-------------------------+
+| Fixed-size byte arrays                         | Allowed                                                 | Allowed                 |
++------------------------------------------------+---------------------------------------------------------+-------------------------+
+| Dynamically-sized byte array                   | Allowed                                                 | Allowed                 |
++------------------------------------------------+---------------------------------------------------------+-------------------------+
+| Enums                                          | Allowed                                                 | Allowed                 |
++------------------------------------------------+---------------------------------------------------------+-------------------------+
+| Fixed-size one dimensional array               | Allowed                                                 |                         |
++------------------------------------------------+---------------------------------------------------------+-------------------------+
+| Dynamically-size one dimensional array         | Allowed                                                 | Allowed                 |
++------------------------------------------------+---------------------------------------------------------+-------------------------+
+| Fixed-size multi-dimensional dimensional array | If you enable the experimental ``ABIEncoderV2`` feature |                         |
++------------------------------------------------+---------------------------------------------------------+-------------------------+
+| Dynamically-size multi-dimensional array       | If you enable the experimental ``ABIEncoderV2`` feature |                         |
++------------------------------------------------+---------------------------------------------------------+-------------------------+
+| ``mapping``                                    | Not allowed                                             | Not allowed             |
++------------------------------------------------+---------------------------------------------------------+-------------------------+
+| struct                                         |                                                         | Only internal functions |
++------------------------------------------------+---------------------------------------------------------+-------------------------+
+
+.. index:: copying types, referencing types
+
+.. _ref-copy-types:
+
+Copying vs referencing
+----------------------
+
+When you use variables as an input or output parameter to functions, how the compiler treats the
+variable depends on its type.
+
+Any variable that is a :ref:`value-types` (fits into 256 bytes), has its value **copied** from the
```

Comment:
`fits into 256 bytes` is not the definition (even though I used this above, yes ;) ).

Reply:
@chriseth The current docs seem to imply this, is there a better definition? Or is it more based on a fairly arbitrary division of types by the team?

Reply:
A value type is a type that is not a reference type and it usually fits 256 bits but not always. Value types are stored on the stack (if they have any data representation at all).

Almost everything apart from structs, arrays and mappings are value types.

Reply:
For some of the weirder types, I guess this is not defined and also does not matter.

Reply:
@chriseth OK, noted for the new PR.